From 88108d2e0329bbd3859dc7991be6c5763a9a548b Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Mon, 14 Oct 2024 12:20:18 -0700
Subject: [PATCH 1/3] ui/gtk: Set 'draw_submitted' only if 'render_sync' is
 false

In case 'render_sync == false', the rendering will start
immediately so it is unnecessary to block further frame submission
with the flag setting (draw_submitted).

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 ui/gtk-egl.c     | 3 ++-
 ui/gtk-gl-area.c | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/ui/gtk-egl.c b/ui/gtk-egl.c
index 7678ab97a5..af7e032d55 100644
--- a/ui/gtk-egl.c
+++ b/ui/gtk-egl.c
@@ -390,8 +390,9 @@ void gd_egl_flush(DisplayChangeListener *dcl,
         gtk_egl_set_scanout_mode(vc, true);
         if (!dmabuf->render_sync) {
             gd_egl_scanout_flush(&vc->gfx.dcl, 0, 0, vc->gfx.w, vc->gfx.h);
+        } else {
+            vc->gfx.guest_fb.dmabuf->draw_submitted = true;
         }
-        vc->gfx.guest_fb.dmabuf->draw_submitted = true;
         gtk_widget_queue_draw_area(area, x, y, w, h);
         return;
     } else if (!dmabuf) {
diff --git a/ui/gtk-gl-area.c b/ui/gtk-gl-area.c
index 64c5cbfa4a..8ebb295249 100644
--- a/ui/gtk-gl-area.c
+++ b/ui/gtk-gl-area.c
@@ -327,9 +327,9 @@ void gd_gl_area_scanout_flush(DisplayChangeListener *dcl,
 		}
             }
 #endif
+        } else {
+            vc->gfx.guest_fb.dmabuf->draw_submitted = true;
         }
-
-        vc->gfx.guest_fb.dmabuf->draw_submitted = true;
         gtk_gl_area_set_scanout_mode(vc, true);
     }
     gtk_gl_area_queue_render(GTK_GL_AREA(vc->gfx.drawing_area));
-- 
2.43.0

